### DISCUSSION TITLE
Annotations generated during clean-up step don't appear in job summary

### DISCUSSION THREAD

**Discussion by jmuollo** [community/community#124176] [Actions]
### Select Topic Area

Product Feedback

### Body

Example:
- a workflow job has a `timeout-minutes` setting on it of 15 minutes 
- the job has a step (1) that runs a process that outputs to a result log as it runs
- the job has a subsequent step (2) that extracts some information from the result log to output as annotations

Sequence of events:
- the job times out after 15 minutes during execution of step 1 - this aborts step 1
- step 2 runs, during the cancellation window, and reads the log that was created during step 1, and outputs some annotations
- however, only the timeout annotations appear in the job summary

The cancellation window is 5 minutes I believe (found [here](https://github.com/orgs/community/discussions/26311#discussioncomment-3251359)) which allows step 2 to run after step 1 is aborted.

The timeout annotations that appear in the job summary are:
```
The job running on runner my_runner has exceeded the maximum execution time of 15 minutes
The operation was canceled.
```
However none of the annotations generated in step 2 appear in the job summary.

I feel like annotations that get generated as part of the clean-up (cancellation) should appear in the job summary.

**Comment by gjabell** [comment]
I'm running into the same issue when using `post` in a custom action to run a cleanup script after the main job fails. If the job fails because of an error, the annotations generated by the `post` script will show up in the summary tab, but if the job times out then they won't. I tried writing to a summary instead as a workaround but this also doesn't show up when the job is cancelled from a timeout.
